pkg/stanza/operator/parser/container/config.go

```diff
@@ -0,0 +1,128 @@
+package container
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+
+	jsoniter "github.com/json-iterator/go"
+	"go.opentelemetry.io/collector/component"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/transformer/recombine"
+)
+
+const operatorType = "container"
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
+}
+
+// NewConfig creates a new JSON parser config with default values
+func NewConfig() *Config {
+	return NewConfigWithID(operatorType)
+}
+
+// NewConfigWithID creates a new JSON parser config with default values
+func NewConfigWithID(operatorID string) *Config {
+	return &Config{
+		ParserConfig:            helper.NewParserConfig(operatorID, operatorType),
+		Format:                  "",
+		AddMetadataFromFilePath: true,
+	}
+}
+
+// Config is the configuration of a Container parser operator.
+type Config struct {
+	helper.ParserConfig `mapstructure:",squash"`
+
+	Format                  string `mapstructure:"format"`
+	AddMetadataFromFilePath bool   `mapstructure:"add_metadata_from_filepath"`
+}
+
+// Build will build a Container parser operator.
+func (c Config) Build(set component.TelemetrySettings) (operator.Operator, error) {
+	parseFromField := entry.NewAttributeField("time")
+	c.ParserConfig.TimeParser = &helper.TimeParser{
+		ParseFrom:  &parseFromField,
+		LayoutType: helper.StrptimeKey,
+		Layout:     "%Y-%m-%dT%H:%M:%S.%LZ",
+	}
+
+	parserOperator, err := c.ParserConfig.Build(set)
+	if err != nil {
+		return nil, err
+	}
+
+	cLogEmitter := helper.NewLogEmitter(set.Logger.Sugar())
+	recombineParser, err := createRecombine(set, cLogEmitter)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create internal recombine config: %w", err)
+	}
+
+	dockerMatcher, err := regexp.Compile(dockerPattern)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create docker regexp: %w", err)
+	}
+	crioMatcher, err := regexp.Compile(crioPattern)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create crio regexp: %w", err)
+	}
+	containerdMatcher, err := regexp.Compile(containerdPattern)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create containerd regexp: %w", err)
+	}
```

Comment:
I think we can use `regexp.MustCompile` and make these global vars, instead of evaluating them here. They will either be valid regex or not at compile time.